c7n/resources/ami.py

```diff
@@ -609,3 +609,72 @@ def process(self, resources, event=None):
                     continue
                 results.extend(f.result())
         return results
+
+
+@AMI.filter_registry.register('image-attribute')
+class ImageAttribute(ValueFilter):
+    """AMI Image Value Filter on a given image attribute.
+
+    Filters AMI's with the given AMI attribute
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: ami-unused-recently
+                resource: ami
+                filters:
+                  - type: image-attribute
+                    attribute: lastLaunchedTime
+                    key: "Value"
```

Comment:
I'm wondering if it's worth simplifying the common case where there's a simple attribute value nested under a `Value` key. Defaulting `self.data['key']` to `Value` during init might be nice :thinking: . Alternatively autoextracting `Value` during fetch... which as of today would work for all attributes except `launchPermissions` and `productCodes`. What do you think - friendly for policy authors or too magical?

(It may also be worth noting that authors should prefer the `cross-account` filter for targeting the `launchPermissions` attribute)

Reply:
This was really the first time I'd played around with the AWS CLI to this level of detail to look at the raw response and map it back to what the policy was doing and in doing so it was actually very clear where the `"Value"` field was coming from.  These just aren't things I do regularly.

On one hand, I agree that defaulting `key` to `"Value"` seems like an easy win and improvement in workflow, on the the other hand I think it's not a terrible thing that it requires the user to do a bit of research to understand the policy they're crafting.

Reply:
Fair point, and less magic is probably a safer default. Any awkwardness of having to specify `key: Value` seems offset by showcasing that in a helpful example (cheers, thanks for that) :+1: